Assets/Scripts/Player/PlayerController.cs

```diff
@@ -111,8 +118,63 @@ private void Update()
             Vector3 playerPosition = transform.position;
             _camera.transform.position = playerPosition + _cameraPosition;
             _camera.transform.LookAt(playerPosition);
+
+            // Check and makes character facing the target object when moving toward it
+            if (target != null) FaceTarget();
+        }
+
+        /// <summary>
+        /// These are functions for character to interact with all objects in the game (ex: moving, following, pick up items, open crate, etc..) using raycast and _NavMeshAgent
+        /// (Won't conflict with character attacking enemy or moving on the map)
+        /// </summary>
+        #region Interactables Functions
+        
+        // This will make character moving to the target object
+        public void MovingToTarget(Interactable newTarget)
+        {
+            _navMeshAgent.stoppingDistance = newTarget.radius * .8f;
+            _navMeshAgent.updateRotation = false;
+            target = newTarget.transform;
+        }
+
+        // Makes character stop moving to target object
+        public void StopMovingToTarget()
+        {
+            _navMeshAgent.stoppingDistance = 0f;
+            _navMeshAgent.updateRotation = true;
+            target = null;
+        }
+
+        // Set focus when interact
+        public void SetFocus(Interactable newFocus)
+        {
+            if(newFocus != focus)
+            {
+                if(focus != null) focus.OnDefocused();
+                focus = newFocus;
+                MovingToTarget(newFocus);
+            }
+
+            newFocus.OnFocused(transform);
         }
 
+        // Remove focus when not interact
+        public void RemoveFocus()
+        {
+            if (focus != null) focus.OnDefocused();
+            focus = null;
+            StopMovingToTarget();
+        }
+
+        // Makes character facing the target when moving toward it
+        void FaceTarget()
```

Comment:
We already have a FaceTarget method. I think you could also just use the existing one :)

Reply:
Okay !